### PR TITLE
feat: update index signatures of 'Option.all' and 'Result.all'

### PR DESCRIPTION
**Pull request by Alx-l**
to only accept an array instead of the spread operator
- update tests accordingly
- reflect updated api inside README

### CODE REVIEW

File: test/option.ts [hunk 1]
```diff
@@ -245,7 +245,7 @@ suite('option', () => {
   })
 
   test('Option.all - 10 Some', () => {
-    const result = Option.all(Some(1), Some('2'), Some(3), Some('4'), Some(5), Some(true), Some(7), Some(8), Some(9), Some(10))
+    const result = Option.all([Some(1), Some('2'), Some(3), Some('4'), Some(5), Some(true), Some(7), Some(8), Some(9), Some(10)])
```
Comment (AlexGalays) [inline test/option.ts]: result is now exactly the same as result2

File: test/result.ts [hunk 2]
```diff
@@ -125,35 +125,35 @@ suite('result', () => {
...
       Err({ error: 'oops' })
-    )
+    ])
     expect(result.isOk()).toBe(false)
     expect(result.get()).toEqual('20')
   })
 
   test('Result.all - 10 Ok', () => {
-    const result = Result.all(Ok(1), Ok('2'), Ok(3), Ok('4'), Ok(5), Ok(true), Ok(7), Ok(8), Ok(9), Ok(10))
+    const result = Result.all([Ok(1), Ok('2'), Ok(3), Ok('4'), Ok(5), Ok(true), Ok(7), Ok(8), Ok(9), Ok(10)])
```
Comment (AlexGalays) [inline test/result.ts]: result is now exactly the same as result2

Comment (AlexGalays) [reply test/result.ts]: too much beer?

Comment (Alx-l) [reply test/result.ts]: Too much work ^^

**Comment by axuan25** [comment]
I'm about to use this method, and so I'd like a little more info on why this change is necessary and/or practical. Would you care to elaborate?

**Comment by AlexGalays** [comment]
Yes, a bit of backgrould would be great. @Alx-l

**Comment by Alx-l** [comment]
I pushed this change after I encountered a problem that wasn't catch by the typescript compiler (and which later would result in an error at runtime).

The problem was caused by the use of the spread operator expected as the argument of the function, which allows you to pass any number of arguments you want, turning in into an array.

So, I was able to pass to the function two arrays (a case which the function wasn't able to process) without being alerted:

```js
const arrA: string[] = []
const arrB: string[] = []

const value = Option.all(arrA, arrB)
  .filter(([maybeArrA, maybeArrB]) =>
    maybeArrA.length > 0 && maybeArrB.length > 2) // Cannot read property 'length' of undefined
  .map(...)
```

And I thought it was a better idea to update the index signature of the function to get closer to the `Promise.all` method (which the name reminds us a lot of both of our functions),
rather than updating the functions to handle this kind of edge case

**Comment by axuan25** [comment]
Thanks @Alx-l ! much appreciated :)

**Comment by AlexGalays** [comment]
This fix is released in 0.7.0

**Pull request merged**